src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -426,7 +429,35 @@ public String getCookie() {
         }
 
         @Override public Collection<? extends SubTask> getSubTasks() {
-            return Collections.singleton(this);
+            List<SubTask> r = new ArrayList<SubTask>();
+            r.add(this);
+            for (int i = 1; i < weight; i++)
+                r.add(new SubTask() {
```

Comment:
I do not think this will serialize correctly. `JenkinsSessionRule` must be used to verify behavior across restarts.

Reply:
Do you mean something like this https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/138/files#diff-649f51b85e49696fd7f4e4fb6ee16c61R1260-R1283?

Reply:
Something like that, using either `RestartableJenkinsRule` or its successor `JenkinsSessionRule`.